include/boost/flags.hpp

```diff
@@ -523,6 +519,88 @@ namespace boost {
+            using value_type = underlying_type;
+
+            BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+                constexpr operator underlying_type() const { return get_underlying(); }
+
+            BOOST_FLAGS_ATTRIBUTE_NODISCARD
+                constexpr underlying_type get_underlying() const { return value.get_underlying(); }
+
+            E value;
+        };
+
+        // class-template to indicate complemented flags
+        // version for complement of enumeration (not nested)
+        template<typename E>
+        struct complement<E,true> {
+            using enumeration_type = enum_type_t<E>;
+            using underlying_type = typename std::underlying_type<enumeration_type>::type;
+            using value_type = underlying_type;
+
+            complement() = default;
+
+            BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+                constexpr complement(value_type v) :
+                value{ v }
+            {}
+
+#if defined(__GNUC__) && (__GNUC__ < 5)
+            // silence (wrong) gcc 4.8 warning "parameter 'v' set but not used" 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wno-unused-but-set-variable"
+#endif // defined(__GNUC__) && (__GNUC__ < 5)
+            BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+                constexpr complement(E v) :
+                value{ static_cast<underlying_type>(v) }
+            {}
+#if defined(__GNUC__) && (__GNUC__ < 5)
+#pragma GCC diagnostic pop
+#endif // defined(__GNUC__) && (__GNUC__ < 5)
+
+            BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+                constexpr operator underlying_type() const { return get_underlying(); }
+
+            BOOST_FLAGS_ATTRIBUTE_NODISCARD
+                constexpr underlying_type get_underlying() const { return value; }
+
+            value_type value;
+        };
+        //// class-template to indicate complemented flags
+        //template<typename E>
+        //struct complement {
+        //    using enumeration_type = typename enum_type_t<E>;
+        //    using underlying_type = typename std::underlying_type<enumeration_type>::type;
+
+        //    complement() = default;
+        //    
+        //    BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+        //        constexpr complement(underlying_type v) :
+        //        value{ v }
+        //    {}
+
+        //    BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+        //        constexpr complement(E v) :
+        //        value{ static_cast<underlying_type>(v) }
+        //    {}
+
+        //    BOOST_FLAGS_ATTRIBUTE_NODISCARD_CTOR
+        //        constexpr operator underlying_type() const { return value; }
+
+        //    // using the underlying_type (instead of enumeration_type or E) ensures that for
+        //    // unscoped enumerations with unspecified underlying type, we
+        //    // don't get into trouble with the "hypothetical value type"
+        //    // cf. https://eel.is/c++draft/dcl.enum#8
+        //    underlying_type value;
+        //};

```

Comment:
Is this dead code? If yes, I'd appreciate if you removed it.

Reply:
dead code is gone